bndtools.core/src/bndtools/central/Central.java

```diff
@@ -212,24 +212,23 @@ public synchronized static Workspace getWorkspace() throws Exception {
             newWorkspace.addBasicPlugin(new WorkspaceListener(newWorkspace));
             newWorkspace.addBasicPlugin(instance.repoListenerTracker);
             newWorkspace.addBasicPlugin(getWorkspaceR5Repository());
+            newWorkspace.addBasicPlugin(new JobProgress());
 
             // Initialize projects in synchronized block
             newWorkspace.getBuildOrder();
 
             // Monitor changes in cnf so we can refresh the workspace
             addCnfChangeListener(newWorkspace);
 
-            // The workspace has been initialized fully, set the field now
-            workspace = newWorkspace;
-
-            workspaceRepositoryChangeDetector = new WorkspaceRepositoryChangeDetector(workspace);
+            workspaceRepositoryChangeDetector = new WorkspaceRepositoryChangeDetector(newWorkspace);
 
-            // Call the queued workspace init callbacks
-            while (!workspaceInitCallbackQueue.isEmpty()) {
-                Function<Workspace,Void> callback = workspaceInitCallbackQueue.remove(0);
-                callback.run(workspace);
+            // The workspace has been initialized fully, set the field now
+            synchronized (instance) {
+                workspace = newWorkspace;
             }
 
+            notifyCallbacks();
```

Comment:
The getWorkspace method is synchronized, so the call here is made holding the Central class monitor. It would be much better to rearrange to call this after the Central class monitor is released.